generate_datafiles/compound_gene_datafiles.py

```diff
@@ -38,30 +46,67 @@
 
 compound_binds_gene_df = pd.read_table(cbg_url, dtype={'entrez_gene_id': int})
 
+compound_regulates_gene_df = (
+    pd.read_table(crg_url, dtype={'entrez_gene_id': int})
+    .assign(sources='lincs')
+    .drop(['z_score', 'status', 'nlog10_bonferroni_pval'], axis=1)
+    .rename(index=str, columns={"perturbagen":'drugbank_id'})
+    )
+
+compound_upregulates_gene_df  = (
+    compound_regulates_gene_df.query("direction == 'up'").drop('direction', axis=1)
+    )
+
+compound_downregulates_gene_df = (
+    compound_regulates_gene_df.query("direction == 'down'").drop('direction', axis=1)
+    )
+
 query = '''
 SELECT "Compound_cid" AS drugbank_id, "Gene_cid" AS entrez_gene_id, count(*) AS n_sentences
 FROM compound_gene
 GROUP BY "Compound_cid", "Gene_cid";
 '''
+
 sentence_count_df = (
     pd.read_sql(query, database_str)
     .astype(dtype={'entrez_gene_id': int})
 )
 
-for r in tqdm_notebook(pd.read_csv("data/compound_gene/compound_binds_gene/compound_gene_pairs_binds_full_map.csv", chunksize=1e6, dtype={'entrez_gene_id': int})):
+for r in tqdm_notebook(pd.read_csv(full_map_output_file, chunksize=1e6, dtype={'entrez_gene_id': int})):
     merged_df = pd.merge(r, compound_binds_gene_df[["drugbank_id", "entrez_gene_id", "sources"]], how="left")
     merged_df['hetionet'] = merged_df.sources.notnull().astype(int)
     merged_df = merged_df.merge(sentence_count_df, how='left', copy=False)
     merged_df.n_sentences = merged_df.n_sentences.fillna(0).astype(int)
     merged_df['has_sentence'] = (merged_df.n_sentences > 0).astype(int)
-    merged_df.to_csv(full_map_sen_count_file, mode='a', index=False)
+    merged_df.to_csv(cbg_sen_count_file, mode='a', index=False)
 
+    merged_df = pd.merge(r, compound_upregulates_gene_df[["drugbank_id", "entrez_gene_id", "sources"]], how="left")
```

Comment:
Looks like a lot of sequential merging steps.  Feels a little worrisome that this is dataset-specific/that there could be an error in one of the steps.  But not aware of an alternative or if this concern is valid

Reply:
Ah yes the reason for the massive set of merging is that I run out of memory handling all the data at once. Turns out taking the Cartesian product of every compound and  every gene breaks 64 GBs of memory. Unlikely there is an error but will test before merge.